internal/rbd/rbd_util.go

```diff
@@ -193,28 +203,95 @@ type migrationVolID struct {
 	clusterID string
 }
 
-var supportedFeatures = map[string]imageFeature{
-	librbd.FeatureNameLayering: {
-		needRbdNbd: false,
-	},
-	librbd.FeatureNameExclusiveLock: {
-		needRbdNbd: false,
-	},
-	librbd.FeatureNameObjectMap: {
-		needRbdNbd: false,
-		dependsOn:  []string{librbd.FeatureNameExclusiveLock},
-	},
-	librbd.FeatureNameFastDiff: {
-		needRbdNbd: false,
-		dependsOn:  []string{librbd.FeatureNameObjectMap},
-	},
-	librbd.FeatureNameJournaling: {
-		needRbdNbd: true,
-		dependsOn:  []string{librbd.FeatureNameExclusiveLock},
-	},
-	librbd.FeatureNameDeepFlatten: {
-		needRbdNbd: false,
-	},
+var (
+	supportedFeatures = map[string]imageFeature{
+		librbd.FeatureNameLayering: {
+			needRbdNbd: false,
+		},
+		librbd.FeatureNameExclusiveLock: {
+			needRbdNbd: false,
+		},
+		librbd.FeatureNameObjectMap: {
+			needRbdNbd: false,
+			dependsOn:  []string{librbd.FeatureNameExclusiveLock},
+		},
+		librbd.FeatureNameFastDiff: {
+			needRbdNbd: false,
+			dependsOn:  []string{librbd.FeatureNameObjectMap},
+		},
+		librbd.FeatureNameJournaling: {
+			needRbdNbd: true,
+			dependsOn:  []string{librbd.FeatureNameExclusiveLock},
+		},
+		librbd.FeatureNameDeepFlatten: {
+			needRbdNbd: false,
+		},
+	}
+
+	krbdLayeringSupport = []util.KernelVersion{
+		{
+			Version:      3,
+			PatchLevel:   8,
+			SubLevel:     0,
+			ExtraVersion: 0,
+			Distribution: "",
+			Backport:     false,
+		},
+	}
+	krbdStripingV2Support = []util.KernelVersion{
+		{
+			Version:      3,
+			PatchLevel:   10,
+			SubLevel:     0,
+			ExtraVersion: 0,
+			Distribution: "",
+			Backport:     false,
+		},
+	}
+	krbdExclusiveLockSupport = []util.KernelVersion{
+		{
+			Version:      4,
+			PatchLevel:   9,
+			SubLevel:     0,
+			ExtraVersion: 0,
+			Distribution: "",
+			Backport:     false,
+		},
+	}
+	krbdDataPoolSupport = []util.KernelVersion{
+		{
+			Version:      4,
+			PatchLevel:   11,
+			SubLevel:     0,
+			ExtraVersion: 0,
+			Distribution: "",
+			Backport:     false,
+		},
+	}
+)
+
+// prepareKrbdFeatureAttrs prepare krbd fearure set based on kernel version.
+// Minimum kernel version should be 3.8, else it will return error.
+// nolint: gocritic // cannot convert if-else to switch.
```

Comment:
please do that, it is a minimal change and prevents needed cleanup later

Reply:
@nixpanic 
I thought we cannot convert this if-else to switch, at least it's not straightforward to me. Can you recheck once?

Reply:
I guess, something like this:

```go
switch {
    case util.CheckKernelSupport(release, krbdDataPoolSupport):
        return librbd.FeatureDataPool, nil
    case ...:
        return ...
}
```

Reply:
Oh, a switch without a condition is possible!! I totally forget that :) Thanks!